src/core/configs/cluster_tests.rs

```diff
@@ -0,0 +1,454 @@
+        assert_eq!(false, d.validate().is_none());
+    }
+
+    #[test]
+    fn test_validate_no_vdisks() {
+        let s = "
+nodes:
+    - name: n1
+      address: 0.0.0.0:111
+      disks:
+        - name: disk1
+          path: /tmp/d1
+";
+        let d: Cluster = YamlConfig {}.parse_config(&s.to_string()).unwrap();
+        assert_eq!(false, d.validate().is_none());
+    }
+
+    #[test]
+    fn test_validate_no_node_with_name() {
+        let s = "
+nodes:
+    - name: other_name
+      address: 0.0.0.0:111
+      disks:
+        - name: disk1
+          path: /tmp/d1
+vdisks:
+    - id: 0
+      replicas:
+        - node: some_name
+          disk: disk1
+";
+        let d: Cluster = YamlConfig {}.parse_config(&s.to_string()).unwrap();
+        assert_eq!(false, d.validate().is_none());
+    }
+
+    #[test]
+    fn test_validate_no_disk_in_node() {
+        let s = "
+nodes:
+    - name: some_name
+      address: 0.0.0.0:111
+      disks:
+        - name: disk1123123123123123
+          path: /tmp/d1
+vdisks:
+    - id: 0
+      replicas:
+        - node: some_name
+          disk: disk1
+";
+        let d: Cluster = YamlConfig {}.parse_config(&s.to_string()).unwrap();
+        assert_eq!(false, d.validate().is_none());
+    }
+
+    #[test]
+    fn test_cluster_convertation() {
+        let s = "
+nodes:
+    - name: n1
+      address: 0.0.0.0:111
+      disks:
+        - name: disk1
+          path: /tmp/d1
+        - name: disk2
+          path: /tmp/d2
+    - name: n2
+      address: 0.0.0.0:1111
+      disks:
+        - name: disk1
+          path: /tmp/d3
+vdisks:
+    - id: 0
+      replicas:
+        - node: n1
+          disk: disk1
+    - id: 1
+      replicas:
+        - node: n1
+          disk: disk2
+        - node: n2
+          disk: disk1
+";
+        let d: Cluster = YamlConfig {}.parse_config(&s.to_string()).unwrap();
+        assert_eq!(true, d.validate().is_none());
+
+        let vdisks = YamlConfig {}.convert_to_data(&d).unwrap();
+        assert_eq!(2, vdisks.len());
+        assert_eq!(0, vdisks[0].id);
+        assert_eq!(1, vdisks[0].replicas.len());
+        assert_eq!("/tmp/d1", vdisks[0].replicas[0].path);
+        assert_eq!(111, vdisks[0].replicas[0].node.port);
+
+        assert_eq!(1, vdisks[1].id);
+        assert_eq!(2, vdisks[1].replicas.len());
+        assert_eq!("/tmp/d2", vdisks[1].replicas[0].path);
+        assert_eq!(111, vdisks[1].replicas[0].node.port);
+        assert_eq!(1111, vdisks[1].replicas[1].node.port);
+    }
+
+    #[test]
+    fn test_ip_parsing() {
+        let s = "
+nodes:
+    - name: n1
+      address: 0.0.0.0:111
+      disks:
+        - name: disk1
+          path: /tmp/d1
+        - name: disk2
+          path: /tmp/d2
+vdisks:
+    - id: 0
+      replicas:
+        - node: n1
+          disk: disk1
+";
+        let d: Cluster = YamlConfig {}.parse_config(&s.to_string()).unwrap();
+        assert_eq!(true, d.validate().is_none());
```

Comment:
Validation will pass with `address: 9999,0:-100000`, and next assertion failes with:
```
'assertion failed: `(left == right)`  left: `111`, right: `31072`
```